### PR TITLE
Season Statistics

### PR DESCRIPTION
**Pull request by ttakahashi1591**
## Description

This PR will add `winningest_coach` and `worst_coach` methods and tests in the `StatTracker` class. It also adds `get_win_percentage`, `get_win_games_from_season`, & `get_lose_games_from_season` methods and tests. 

## Type of change

- [ ] fix
- [x] feat
- [x] test
- [x] refactor
- [ ] docs

## Checklist

- [x] code has been self reviewed
- [x] code runs without any errors
- [x] thorough testing has been implemented if adding feature
- [x] all tests pass

### Thanks!

### CODE REVIEW

File: lib/stat_tracker.rb
Comment (lanebret85) [inline lib/stat_tracker.rb:1]: good start on the stat tracker!

**Review by aevans27 — APPROVED** [review]
Checked the conflicts, looks good, will run tests to make sure conflicts didn't change too much.

File: lib/team_result_parser.rb
Comment (lanebret85) [inline lib/team_result_parser.rb:1]: methods look good here!

File: spec/team_result_parser_spec.rb
Comment (lanebret85) [inline spec/team_result_parser_spec.rb:1]: nice job on the tests! layout looks good

**Review by lanebret85 — APPROVED** [review]

**Pull request merged**